sftp/sftp.go

```diff
@@ -101,6 +94,81 @@ func (fm *fileManagerImpl) Delete(remoteFilePath string) error {
 	if err != nil {
 		return fmt.Errorf("cannot delete file: %w", err)
 	}
+	return nil
+}
 
+func (fm *fileManagerImpl) Reset() error {
```

Comment:
Should we expose `Reset` method, and why is it necessary to create a new client every time we want to retry?

Reply:
sftp server will throw the` connection lost` error for idle connections. In Reset method we are creating a new client with same config.

Reply:
Wouldn't it make sense to wrap the client in our own to encapsulate this behaviour and make it transparent to the end user of this library?

I also find confusing that inside `*fileManagerImpl.Reset()` you're using the constructor to build another `**fileManagerImpl` but you're discarding everything else and keeping the client. Seems like an anti-pattern.

Also the constructor is returning an unexported type. Might as well return the interface there since you have exported it.

Reply:
Updated the Reset() method and used newSFTPClientFromConfig() for sftp client creation instead of using file manager constructor.